### ISSUE TITLE
VideoRedo TVS edits and CCE

### ISSUE THREAD

**Issue by yrp888** [CCExtractor/ccextractor#341]
Australian DVB-T Topfield recordings show captions on Topfield and VRD but CCE does not extract captions after VRD edit.

Topfield mpeg file edited in VRD and saved as TS. VRD identifies subtitle info in TS and displays captions while playing so captions are passed through to cut file but CCE cannot extract them.

Sample file with VRD info and CCE activity log -
(link removed)

**Comment by bigharshrag** [comment]
@yrp888 It appears your link doesn't work. Could you check it, please?

**Comment by yrp888** [comment]
Sorry, newb error, I should have previewed the post and checked the link.

I've reissued the link.
https://www.dropbox.com/s/5w64fztwuujxlm8/VideoEdit.zip?dl=0

@cfsmp3 has requested an unedited source file to compare to. The file is large and will take some time to upload and I'll post that link once complete.

**Comment by yrp888** [comment]
New link with both source and cut files
https://www.dropbox.com/s/ynnyswy8hnepugj/VideoEdit.zip?dl=0

**Comment by grave-w-grave** [comment]
Hey, when I try to run ccextractor on both sample files, it just crashes. Windows 10 and v. 0.82.

Console output:
`ccextractorwin "MSWsource-01.mpg"
CCExtractor 0.82, Carlos Fernandez Sanz, Volker Quetschke.
Teletext portions taken from Petr Kutalek's telxcc
--------------------------------------------------------------------------
Input: MSWsource-01.mpg
[Extract: 1] [Stream mode: Autodetect]
[Program : Auto ] [Hauppage mode: No] [Use MythTV code: Auto]
[Timing mode: Auto] [Debug: No] [Buffer input: Yes]
[Use pic_order_cnt_lsb for H.264: No] [Print CC decoder traces: No]
[Target format: .srt] [Encoding: UTF-8] [Delay: 0] [Trim lines: No]
[Add font color data: Yes] [Add font typesetting: Yes]
[Convert case: No] [Video-edit join: No]
[Extraction start time: not set (from start)]
[Extraction end time: not set (to end)]
[Live stream: No] [Clock frequency: 90000]
[Teletext page: Autodetect]
[Start credits text: None]

-----------------------------------------------------------------
Opening file: MSWsource-01.mpg
Detected MP4 box with name: meco
File seems to be a transport stream, enabling TS mode
Analyzing data in general mode

This TS file has more than one program. These are the program numbers found:
544
545
546
547
548
550
551
VBI/teletext stream ID 576 (0x240) for SID 545 (0x221)
Creating MSWsource-01.srt
Notice: Teletext page with possible subtitles detected: 801
- No teletext page specified, first received suitable page is 801, not guaranteed`

`ccextractorwin "MSW_VRDcut-01.ts"
CCExtractor 0.82, Carlos Fernandez Sanz, Volker Quetschke.
Teletext portions taken from Petr Kutalek's telxcc
--------------------------------------------------------------------------
Input: MSW_VRDcut-01.ts
[Extract: 1] [Stream mode: Autodetect]
[Program : Auto ] [Hauppage mode: No] [Use MythTV code: Auto]
[Timing mode: Auto] [Debug: No] [Buffer input: Yes]
[Use pic_order_cnt_lsb for H.264: No] [Print CC decoder traces: No]
[Target format: .srt] [Encoding: UTF-8] [Delay: 0] [Trim lines: No]
[Add font color data: Yes] [Add font typesetting: Yes]
[Convert case: No] [Video-edit join: No]
[Extraction start time: not set (from start)]
[Extraction end time: not set (to end)]
[Live stream: No] [Clock frequency: 90000]
[Teletext page: Autodetect]
[Start credits text: None]

-----------------------------------------------------------------
Opening file: MSW_VRDcut-01.ts
File seems to be a transport stream, enabling TS mode
Analyzing data in general mode
VBI/teletext stream ID 576 (0x240) for SID 545 (0x221)
Creating MSW_VRDcut-01.srt
Notice: Teletext page with possible subtitles detected: 801
- No teletext page specified, first received suitable page is 801, not guaranteed
  1%  |  00:05`

**Comment by cfsmp3** [comment]
It is an actual crash (with a nasty windows popup and all), not just
program termination?

**Comment by grave-w-grave** [comment]
Yeah, windows gives a "ccextractor has stopped working" popup.

**Comment by cfsmp3** [comment]
OK - obviously issue remains open until solved.

**Comment by grave-w-grave** [comment]
Fixed as of 0.84: [Debug.zip](https://github.com/CCExtractor/ccextractor/files/691869/Debug.zip)

```
CCExtractor 0.84, Carlos Fernandez Sanz, Volker Quetschke.
Teletext portions taken from Petr Kutalek's telxcc
--------------------------------------------------------------------------
Input: MSWsource-01.mpg
[Extract: 1] [Stream mode: Autodetect]
[Program : Auto ] [Hauppage mode: No] [Use MythTV code: Auto]
[Timing mode: Auto] [Debug: No] [Buffer input: Yes]
[Use pic_order_cnt_lsb for H.264: No] [Print CC decoder traces: No]
[Target format: .srt] [Encoding: UTF-8] [Delay: 0] [Trim lines: No]
[Add font color data: Yes] [Add font typesetting: Yes]
[Convert case: No] [Video-edit join: No]
[Extraction start time: not set (from start)]
[Extraction end time: not set (to end)]
[Live stream: No] [Clock frequency: 90000]
[Teletext page: Autodetect]
[Start credits text: None]

-----------------------------------------------------------------
Opening file: MSWsource-01.mpg
Detected MP4 box with name: meco
File seems to be a transport stream, enabling TS mode
Analyzing data in general mode

This TS file has more than one program. These are the program numbers found:
544
545
546
547
548
550
551
VBI/teletext stream ID 576 (0x240) for SID 545 (0x221)
Creating MSWsource-01.srt
Notice: Teletext page with possible subtitles detected: 801
- No teletext page specified, first received suitable page is 801, not guaranteed
 99%  |  00:00Premature end of file - Transport Stream packet is incomplete (expected 188 bytes, got 108).
100%  |  00:00
Teletext decoder: 25203 packets processed

Number of NAL_type_7: 0
Number of VCL_HRD: 0
Number of NAL HRD: 0
Number of jump-in-frames: 0
Number of num_unexpected_sei_length: 0
Done, processing time = 1 seconds
This is beta software. Report issues to carlos at ccextractor org...
CCExtractor 0.84, Carlos Fernandez Sanz, Volker Quetschke.
Teletext portions taken from Petr Kutalek's telxcc
--------------------------------------------------------------------------
Input: MSWsource-01.mpg
[Extract: 1] [Stream mode: Autodetect]
[Program : Auto ] [Hauppage mode: No] [Use MythTV code: Auto]
[Timing mode: Auto] [Debug: No] [Buffer input: Yes]
[Use pic_order_cnt_lsb for H.264: No] [Print CC decoder traces: No]
[Target format: .srt] [Encoding: UTF-8] [Delay: 0] [Trim lines: No]
[Add font color data: Yes] [Add font typesetting: Yes]
[Convert case: No] [Video-edit join: No]
[Extraction start time: not set (from start)]
[Extraction end time: not set (to end)]
[Live stream: No] [Clock frequency: 90000]
[Teletext page: Autodetect]
[Start credits text: None]

-----------------------------------------------------------------
Opening file: MSWsource-01.mpg
Detected MP4 box with name: meco
File seems to be a transport stream, enabling TS mode
Analyzing data in general mode

This TS file has more than one program. These are the program numbers found:
544
545
546
547
548
550
551
VBI/teletext stream ID 576 (0x240) for SID 545 (0x221)
Creating MSWsource-01.srt
Notice: Teletext page with possible subtitles detected: 801
- No teletext page specified, first received suitable page is 801, not guaranteed
 99%  |  00:00Premature end of file - Transport Stream packet is incomplete (expected 188 bytes, got 108).
100%  |  00:00
Teletext decoder: 25203 packets processed

Number of NAL_type_7: 0
Number of VCL_HRD: 0
Number of NAL HRD: 0
Number of jump-in-frames: 0
Number of num_unexpected_sei_length: 0
Done, processing time = 1 seconds
This is beta software. Report issues to carlos at ccextractor org...

```

```
CCExtractor 0.84, Carlos Fernandez Sanz, Volker Quetschke.
Teletext portions taken from Petr Kutalek's telxcc
--------------------------------------------------------------------------
Input: MSW_VRDcut-01.ts
[Extract: 1] [Stream mode: Autodetect]
[Program : Auto ] [Hauppage mode: No] [Use MythTV code: Auto]
[Timing mode: Auto] [Debug: No] [Buffer input: Yes]
[Use pic_order_cnt_lsb for H.264: No] [Print CC decoder traces: No]
[Target format: .srt] [Encoding: UTF-8] [Delay: 0] [Trim lines: No]
[Add font color data: Yes] [Add font typesetting: Yes]
[Convert case: No] [Video-edit join: No]
[Extraction start time: not set (from start)]
[Extraction end time: not set (to end)]
[Live stream: No] [Clock frequency: 90000]
[Teletext page: Autodetect]
[Start credits text: None]

-----------------------------------------------------------------
Opening file: MSW_VRDcut-01.ts
File seems to be a transport stream, enabling TS mode
Analyzing data in general mode
VBI/teletext stream ID 576 (0x240) for SID 545 (0x221)
Creating MSW_VRDcut-01.srt
Notice: Teletext page with possible subtitles detected: 801
- No teletext page specified, first received suitable page is 801, not guaranteed
100%  |  08:09
Teletext decoder: 24494 packets processed

Number of NAL_type_7: 0
Number of VCL_HRD: 0
Number of NAL HRD: 0
Number of jump-in-frames: 0
Number of num_unexpected_sei_length: 0

Min PTS:                                00:00:00:240
Max PTS:                                00:00:00:240
Length:                          00:00:00:000
Done, processing time = 0 seconds
This is beta software. Report issues to carlos at ccextractor org...
```

**Comment by yrp888** [comment]
fantastic, thanks guys for investigating and fixing. Will download the update now.
